### PR TITLE
Clean up LecturesDatabaseRepository

### PR DESCRIPTION
**Pull request by cketti**
- Avoid creating lots of temporary objects
- ~~Simplify code to iterate over a Cursor~~

### CODE REVIEW

**Comment by johnjohndoe** [comment]
@cketti As told in person I will take a look at your contribution after 36C3 as soon as I have time for it. Thank you for the effort.

File: database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/LecturesDatabaseRepository.kt [hunk 1]
```diff
@@ -84,46 +84,48 @@ class LecturesDatabaseRepository(
...
+                    changedIsNew = cursor.getInt(CHANGED_IS_NEW).isChanged,
+                    changedLanguage = cursor.getInt(CHANGED_LANGUAGE).isChanged,
+                    changedRecordingOptOut = cursor.getInt(CHANGED_RECORDING_OPTOUT).isChanged,
+                    changedRoom = cursor.getInt(CHANGED_ROOM).isChanged,
+                    changedSpeakers = cursor.getInt(CHANGED_SPEAKERS).isChanged,
+                    changedSubtitle = cursor.getInt(CHANGED_SUBTITLE).isChanged,
+                    changedTime = cursor.getInt(CHANGED_TIME).isChanged,
+                    changedTitle = cursor.getInt(CHANGED_TITLE).isChanged,
+                    changedTrack = cursor.getInt(CHANGED_TRACK).isChanged
+            )
```
Comment (johnjohndoe) [inline database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/LecturesDatabaseRepository.kt]: Please also apply this pattern to the repositories (`AlarmsDatabaseRepository#query`, `MetaDatabaseRepository#query`) which also use the `copy` method to have a consistent code base.

File: database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/LecturesDatabaseRepository.kt [hunk 2]
```diff
@@ -82,8 +82,7 @@ class LecturesDatabaseRepository(
             return lectures.toList()
         }
 
-        cursor.moveToFirst()
-        while (!cursor.isAfterLast) {
+        while (cursor.moveToNext()) {
```
Comment (johnjohndoe) [inline database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/LecturesDatabaseRepository.kt]: Please also apply this pattern to the other repositories to have a consistent code base.

Comment (cketti) [reply database/src/main/java/info/metadude/android/eventfahrplan/database/repositories/LecturesDatabaseRepository.kt]: I reverted this change and will create a new pull request for that pattern.

**Comment by johnjohndoe** [comment]
# Successfully tested on
with `ccc36c3` flavor
- :heavy_check_mark: Google Pixel 2, Android 10 (API 29)

**Pull request merged**